modules/account-lib/src/coin/stacks/utils.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  AddressHashMode,
+  StacksTransaction,
+  TransactionVersion,
+  addressFromVersionHash,
+  addressHashModeToVersion,
+  addressToString,
+  validateStacksAddress,
+} from '@stacks/transactions';
+import BigNumber from 'bignumber.js';
+
+/**
+ * Encodes a buffer as a `0x` prefixed lower-case hex string.
+ *
+ * @param {Buffer} buff - a buffer with a hexadecimal string
+ * @returns {string} - the hexadecimal string prefixed with "0x"
+ */
+export function bufferToHexPrefixString(buff: Buffer): string {
+  return '0x' + buff.toString('hex');
+}
+
+/**
+ * @param hex
+ */
+export function removeHexPrefix(hex: string): string {
+  if (hex.startsWith('0x')) return hex.slice(2);
+  else return hex;
+}
+
+/**
+ * @param publicKeyHash
+ * @param hashMode
+ * @param transactionVersion
+ */
+function getAddressFromPublicKeyHash(
+  publicKeyHash: Buffer,
+  hashMode: AddressHashMode,
+  transactionVersion: TransactionVersion,
+): string {
+  const addrVer = addressHashModeToVersion(hashMode, transactionVersion);
+  if (publicKeyHash.length !== 20) {
+    throw new Error('expected 20-byte pubkeyhash');
+  }
+  const addr = addressFromVersionHash(addrVer, publicKeyHash.toString('hex'));
+  const addrString = addressToString(addr);
+  return addrString;
+}
+
+/**
+ * @param tx
+ */
+export function getTxSenderAddress(tx: StacksTransaction): string {
+  if (tx.auth.spendingCondition !== null && tx.auth.spendingCondition !== undefined) {
+    const spendingCondition = tx.auth.spendingCondition;
+    const txSender = getAddressFromPublicKeyHash(
+      Buffer.from(spendingCondition.signer, 'hex'),
+      spendingCondition.hashMode as number,
+      tx.version,
+    );
+    return txSender;
+  } else throw new Error('spendingCondition should not be null');
+}
+
+/**
+ * Returns whether or not the string is a valid amount number
+ *
+ * @param {string} amount - the string to validate
+ * @returns {boolean} - the validation result
+ */
+export function isValidAmount(amount: string): boolean {
+  const bigNumberAmount = new BigNumber(amount);
+  return bigNumberAmount.isInteger() && bigNumberAmount.isGreaterThanOrEqualTo(0);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol address
+ *
+ * @param {string} address - the address to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidAddress(address: string): boolean {
+  return validateStacksAddress(address);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol transaction id or not
+ *
+ * @param {string} txId - the transaction id to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidTransactionId(txId: string): boolean {
```

Comment:
source

Reply:
This does not appear to be documented in SIP-005.  However, the code to generate a transaction id is in the `txid()` method in the [StacksTransaction](https://github.com/blockstack/stacks.js/blob/master/packages/transactions/src/transaction.ts#L176) class.  It performs a SHA-512/256 [hash](https://github.com/blockstack/stacks.js/blob/3f889aab44fa62c480258213a920e30f29833c2a/packages/transactions/src/utils.ts#L97) of the serialized transaction.

Reply:
I've added a comment for this.